consensus/obcpbft/viewchange.go

```diff
@@ -413,6 +413,8 @@ func (instance *pbftCore) processNewView2(nv *NewView) error {
 		instance.persistQSet()
 	}
 
+	instance.updateViewChangeSeqNo()
```

Comment:
Could you explain the logic here?  In particular, how can you be certain that `instance.seqNo` is consistent across all replicas at this point?

Maybe the `if` statement above isn't as conditional as it seems, and could be clarified instead to be follow the loop as something like `instance.seqNo = max(xSet)`, but it is not immediately obvious to me.

What if a primary is partitioned from the network, advances its own `instance.seqNo` from n to n+3, but the network never receives pre-prepares for n+1, n+2, or n+3, when the view change completes, the highest sequence number in the xSet is n+1.  Now, the old primary will set its next view change mark to be (n+3)+instance.viewChangePeriod, while the rest of the network picks (n+1)+instance.viewChangePeriod.

This may be a more general bug of how we manage instance.seqNo, but since this had only previously been used to assign sequence numbers to pre-prepares for the primary (and the primary was the one computing the xSet) it may not have been a problem that it was potentially out of sync across replicas on view change.

I don't think this is likely to deadlock the network, but it could cause timeout induced view changes, and degrade the network performance.